monkey/infection_monkey/credential_collectors/credential_components/i_credential_component.py

```diff
@@ -0,0 +1,10 @@
+from abc import ABC
+from dataclasses import dataclass
+
+from ..credential_types import CredentialTypes
+
+
+@dataclass
+class ICredentialComponent(ABC):
```

Comment:
The subtypes make more sense if they validate/enforce the type of data they store. For example, the `KeyPair` constructor should take `public_key` and `private_key` as arguments and populate `contents`. the NT and LM hash classes should validate the format of their input. etc.

Reply:
We don't explicitly implement any methods here. IIRC dataclass doesn't add any new methods, it just modifies __init__ and __repr__. If we agree that interfaces express vanilla object behavior then I'll rename. If we agree that interfaces have vanilla object interface, than there's no need to rename to `AbstractCredentialComponent`.

Reply:
> For example, the KeyPair constructor should take public_key and private_key as arguments and populate contents.

I'm not sure what changes you reviewed, but it's being done apart from validation. Validation should probably be done by collectors, not data structures.

Contents encapsulate data, everything else is metadata about the data-structure (in this case only type). I don't think this is a big difference, except that credential component won't be able to have `type`, since it's already reserved